docs/sources/experiments/cohort-labels.md

```diff
@@ -0,0 +1,493 @@
+
+
+```
+config_version: 'v6'
+
+temporal_config:
+    feature_start_time: '2010-01-04'
+    feature_end_time: '2018-03-01'
+    label_start_time: '2015-02-01'
+    label_end_time: '2018-03-01'
+
+    model_update_frequency: '1y'
+    training_label_timespans: ['1month']
+    training_as_of_date_frequencies: '1month'
+
+    test_durations: '1month'
+    test_label_timespans: ['1month']
+    test_as_of_date_frequencies: '1month'
+
+    max_training_histories: '5y'
+
+cohort_config:
+  query: |
+    select distinct(entity_id)
+    from permits
+    where
+    tsrange(start_time, end_time, '[]') @> {as_of_date}
+  name: 'permits_in_last_year'
+
+label_config:
+  query: |
+    select
+    entity_id,
+    bool_or(result = 'fail')::integer as outcome
+    from inspections
+    where '{as_of_date}'::timestamp <= date
+    and date < '{as_of_date}'::timestamp + interval '{label_timespan}'
+    group by entity_id
+  name: 'failed_inspection'
+
+feature_aggregations:
+    -
+        prefix: 'test'
+        from_obj: 'permits'
+        knowledge_date_column: 'date'
+        aggregates_imputation:
+            all:
+                type: 'zero_noflag'
+        aggregates: [{quantity: '1', metrics: ['sum']}]
+        intervals: ['3month']
+        groups: ['entity_id']
+```
+
+The above feature aggregation should just create a feature with the value `1` for each entity, but what's important here is that it's a valid feature config that allows us to make complete matrices. To make matrices using all of this configuration, you can run:
+
+```python
+from triage.experiments import SingleThreadedExperiment
+from triage import create_engine
+import yaml
+
+with open('<your_experiment_config.yaml>') as fd:
+    experiment_config = yaml.load(fd)
+experiment = SingleThreadedExperiment(
+    experiment_config=experiment_config,
+    db_engine=create_engine(...),
+    project_path='./'
+)
+experiment.generate_matrices()
+```
+
+The matrix generation process will run all of the cohort/label/feature generation above, and then save matrices to your project_path's `matrices` directory. By default, these are CSVs and should have a few columns: 'entity_id', 'date', 'test_1_sum', and 'failed_inspection'. The 'entity_id' and 'date' columns represent the index of this matrix, and 'failed_inspection' is the label.  Each of these CSV files has a YAML file starting with the same hash representing metadata about that matrix. If you want to look for just the train matrices to inspect the results of the `include_missing_labels_in_train_as` flag, try this command (assuming you can use bash):
```

Comment:
It will be nice to include some sentences about how the matrix name is created (i.e. their dependencies) and how this will be stable.

Reply:
I could add this, but is this in scope for a document that's focused on cohorts and labels? The only reason I even added anything about matrices is that that's where the joins and the label imputation is visible. 

The 'Restarting an Experiment' section of the running document does cover this, though it could go into more detail for sure. 

https://github.com/dssg/triage/blob/master/docs/sources/experiments/running.md#restarting-an-experiment

Is that a better place to put those details or do you think this cohort/label document makes more sense?

Reply:
You are right :)